src/Cli/func/Helpers/AzureHelper.cs

```diff
@@ -383,31 +383,40 @@ internal static async Task CheckFunctionHostStatusForFlex(Site functionApp, stri
             functionAppReadyClient.DefaultRequestHeaders.Add("User-Agent", Constants.CliUserAgent);
             functionAppReadyClient.DefaultRequestHeaders.Add("Accept", jsonContentType);
 
+            var statusUri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+            ColoredConsole.WriteLine($"Target endpoint: {statusUri.GetLeftPart(UriPartial.Path)}");
+
             await RetryHelper.Retry(
                 async () =>
                 {
+                    // Remove the header first if it already exists from a previous retry
+                    functionAppReadyClient.DefaultRequestHeaders.Remove("x-ms-request-id");
                     functionAppReadyClient.DefaultRequestHeaders.Add("x-ms-request-id", Guid.NewGuid().ToString());
-                    var uri = new Uri($"https://{functionApp.HostName}/admin/host/status?code={masterKey}");
+
                     var request = new HttpRequestMessage()
                     {
-                        RequestUri = uri,
+                        RequestUri = statusUri,
                         Method = HttpMethod.Get
                     };
 
-                    var response = await functionAppReadyClient.SendAsync(request);
-                    ColoredConsole.Write(".");
+                    HttpResponseMessage response = await functionAppReadyClient.SendAsync(request);
+                    string responseContent = await response.Content.ReadAsStringAsync();
 
                     if (response.IsSuccessStatusCode)
                     {
-                        ColoredConsole.WriteLine(" done");
+                        ColoredConsole.WriteLine($"Host status check succeeded with response: {responseContent}");
```

Comment:
Undo this to `ColoredConsole.WriteLine(" done");`

Log `responseContent` in a new line or log statement